spec/controllers/application_controller_spec.rb

```diff
@@ -0,0 +1,163 @@
+require 'spec_helper'
+
+describe ApplicationController do
+
+
+  it 'loads the homepage' do 
+    get '/'
+    expect(last_response.status).to eq(200)
+    expect(last_response.body).to include("Welcome to Fwitter")
+  end
+
+  it 'loads the signup page' do 
+    get '/signup'
+    expect(last_response.status).to eq(200)
+  end
+
+  it 'signup directs user to twitter index' do 
+    params = {
+      :username => "skittles123",
+      :email => "skittles@aol.com",
+      :password => "rainbows"
+    }
+    post '/signup', params
+    expect(last_response.location).to eq("http://example.org/tweets") 
+  end
+
+  it 'loads the login page' do
+    get '/login' 
+    expect(last_response.status).to eq(200)
+  end
+
+  it 'loads the tweets index after login' do
+    user = User.create(:username => "becky567", :email => "starz@aol.com", :password => "kittens")
+    params = {
+      :username => "becky567",
+      :password => "kittens"
+    }
+    post '/login', params
+    expect(last_response.status).to eq(302)
+    expect(last_response.location).to eq("http://example.org/tweets")
+  end
+
+  
+  it 'redirects to login after logout' do 
+    get '/logout' 
+    expect(last_response.location).to eq("http://example.org/login")
+  end
+
+  it 'does not load /tweets if user not logged in' do 
+    get '/tweets'
+    expect(last_response.location).to eq("http://example.org/login")
+  end 
+
+  it 'does load /tweets if user is logged in' do
+    user = User.create(:username => "becky567", :email => "starz@aol.com", :password => "kittens")
+
+    visit '/login'
+
+    fill_in(:username, :with => "becky567")
```

Comment:
This is a great use case for contexts. Effectively it's the same tests, but different contexts (logged in/logged out) expect different responses